src/pages/db/_components/use-cases/Reactions.tsx

```diff
@@ -0,0 +1,43 @@
+import { createSignal } from "solid-js"
+
+export default function Reactions() {
+	const [selected, setSelected] = createSignal(false)
+	const [reactions, setReactions] = createSignal([
+		{ id: 1, emoji: "🔥", count: 1121, isClicked: false },
+		{ id: 2, emoji: "👍", count: 459, isClicked: false },
+		{ id: 3, emoji: "👎", count: 391, isClicked: false },
+		{ id: 4, emoji: "🚀", count: 275, isClicked: false },
+		{ id: 5, emoji: "🎉", count: 137, isClicked: false }
+	])
+
+	function handleReactionClick(id: number) {
+		const updatedReactions = reactions().map((reaction) => {
+			if(selected() === true && reaction.id === id){
+				setSelected(false)
```

Comment:
Calling state setters (or doing any side effects) in a `.map` callback is potentially confusing; `.map` callbacks should ideally be pure.